manifests/repo/rpm.pp

```diff
@@ -1,6 +1,9 @@
 # @summary Manage Open OnDemand RPM repos
 # @api private
-class openondemand::repo::rpm {
+class openondemand::repo::rpm (
+  Integer $gpgcheck      = 1,
+  Integer $repo_gpgcheck = 1,
```

Comment:
I'd recommend moving these to `manifests/init.pp` so that all class parameters are in one place.

Reply:
Also make the data type at least support `Enum['1','0']` or further support more per what yumrepo supports: https://github.com/puppetlabs/puppetlabs-yumrepo_core/blob/main/lib/puppet/type/yumrepo.rb#L25